packages/radix-ui-themes/src/theme-panel.tsx

```diff
@@ -621,5 +638,30 @@ function upperFirst(string: string) {
   return string.charAt(0).toUpperCase() + string.slice(1);
 }
 
+function updateRootAppearanceClass(appearance: 'light' | 'dark') {
+  const root = document.documentElement;
+  const hasLightTheme = root.classList.contains('light-theme');
+  const hasDarkTheme = root.classList.contains('dark-theme');
+  const hasLight = root.classList.contains('light');
+  const hasDark = root.classList.contains('dark');
+
+  if (hasLightTheme || hasDarkTheme) {
+    root.classList.remove('light-theme', 'dark-theme');
+    root.style.colorScheme = appearance;
+    root.classList.add(`${appearance}-theme`);
+  }
+
+  if (hasLight || hasDark) {
+    root.classList.remove('light', 'dark');
+    root.style.colorScheme = appearance;
+    root.classList.add(appearance);
+  }
+
+  if (!hasLightTheme && !hasDarkTheme && !hasLight && !hasDark) {
+    root.style.colorScheme = appearance;
+    root.classList.add(appearance);
+  }
+}
```

Comment:
I went the other way around @benoitgrelard here...

Before, when using the appearance theme toggle in the theme panel when initial `appearance` was `inherit`, it was forced into `light`/`dark`.

Instead, I'm now preserving the original `appearance="inherit"` prop and toggling the class on the html element, and the prop value is changed only when an explicit `appearance` value is used.